### PR TITLE
ARROW-13671: [Dev] Fix conda recipe on Arm 64k page system

### PR DESCRIPTION
**Pull request by cyb70289**

### CODE REVIEW

**Comment by cyb70289** [comment]
@kszucs @xhochy 

Not tested yet. Any easy way to build and test?

**Comment by cyb70289** [comment]
@github-actions crossbow submit conda-linux-gcc-py38-arm64

**Comment by cyb70289** [comment]
I've validated this PR on both 64k and 4k page arm64 system. It works correctly.
I also tried night build conda image, it fails on 64k page system.

File: dev/tasks/conda-recipes/arrow-cpp/build-arrow.sh [hunk 1]
```diff
@@ -43,6 +43,12 @@ else
     EXTRA_CMAKE_ARGS=" ${EXTRA_CMAKE_ARGS} -DARROW_GANDIVA=ON"
 fi
 
+if [[ "${target_platform}" == "linux-aarch64" ]]; then
+    # To support both 4k and 64k page arm64 systems
+    # See https://github.com/apache/arrow/pull/10940
+    EXTRA_CMAKE_ARGS=" ${EXTRA_CMAKE_ARGS} -DARROW_JEMALLOC_LG_PAGE=16"
```
Comment (xhochy) [inline dev/tasks/conda-recipes/arrow-cpp/build-arrow.sh:49]: Can you remove the `sed` workaround above for `osx-arm64`?
```suggestion
    EXTRA_CMAKE_ARGS=" ${EXTRA_CMAKE_ARGS} -DARROW_JEMALLOC_LG_PAGE=16"
elif [[ "${target_platform}" == "osx-arm64" ]]; then
    EXTRA_CMAKE_ARGS=" ${EXTRA_CMAKE_ARGS} -DARROW_JEMALLOC_LG_PAGE=14"
```

**Comment by cyb70289** [comment]
@github-actions crossbow submit --help

File: dev/tasks/conda-recipes/arrow-cpp/build-arrow.sh (hunk 1 above)
Comment (cyb70289) [reply dev/tasks/conda-recipes/arrow-cpp/build-arrow.sh]: Done

**Comment by cyb70289** [comment]
@github-actions crossbow submit conda-linux-gcc-py38-arm64

**Comment by cyb70289** [comment]
@github-actions crossbow submit conda-osx-arm64-clang-py39

**Comment by kszucs** [comment]
@github-actions crossbow submit conda-osx-arm64-clang-py39

**Comment by cyb70289** [comment]
@github-actions crossbow submit conda-osx-arm64-clang-py39

**Comment by cyb70289** [comment]
@kszucs @xhochy I've rebased. OSX test passed now.

**Review by kszucs — APPROVED** [review]
Thanks @cyb70289!

**Pull request closed without merging**